filesystems/watcher_test.go

```diff
@@ -206,73 +206,49 @@ func TestDoubleUpload(t *testing.T) { // nolint: gocyclo
 
 	shouldRotate = CheckFileForRotation("stdout.2", nil)
-	if !shouldRotate {
-		t.Fatal("stdout.2 should rotate")
-	}
+	assert.True(t, shouldRotate, "stdout.2 should rotate")
 
 	shouldRotate = CheckFileForRotation("stout.2", nil)
-	if shouldRotate {
-		t.Fatal("stout.2 should not rotate")
-	}
+	assert.False(t, shouldRotate, "stout.2 should not rotate")
 
 	shouldRotate = CheckFileForRotation("stdout.30092016_151555.073", nil)
-	if !shouldRotate {
-		t.Fatal("stdout.30092016_151555.073 should rotate")
-	}
+	assert.True(t, shouldRotate, "stdout.30092016_151555.073 should rotate")
 
 	shouldRotate = CheckFileForRotation("stderr.70092016_151555.07888", nil)
-	if !shouldRotate {
-		t.Fatal("stderr.70092016_151555.07888 should rotate")
-	}
+	assert.True(t, shouldRotate, "stderr.70092016_151555.07888 should rotate")
 
 	shouldRotate = CheckFileForRotation("stderr.foobar.4555.", nil)
-	if shouldRotate {
-		t.Fatal("stderr.foobar.4555 should not rotate")
-	}
+	assert.False(t, shouldRotate, "stderr.foobar.4555 should not rotate")
 
 	shouldRotate = CheckFileForRotation("stderr.70092016_151555.07888.backup", nil)
-	if shouldRotate {
-		t.Fatal("stderr.70092016_151555.07888.backup should not rotate")
-	}
+	assert.False(t, shouldRotate, "stderr.70092016_151555.07888.backup should not rotate")
 
 	allLogsRegexp := regexp.MustCompile(`^[\w \d \. _ -]*log$`)
 	shouldRotate = CheckFileForRotation("propsapi_i-4a7ec3d1_tomcat_catalina.out_20160329_1352.log", allLogsRegexp)
-
-	if !shouldRotate {
-		t.Fatal("propsapi_i-4a7ec3d1_tomcat_catalina.out_20160329_1352.log should rotate")
-	}
+	assert.True(t, shouldRotate, "propsapi_i-4a7ec3d1_tomcat_catalina.out_20160329_1352.log should rotate")
 
 	shouldRotate = CheckFileForRotation("stderr.20", allLogsRegexp)
-	if !shouldRotate {
-		t.Fatal("stderr.20 should rotate")
-	}
+	assert.True(t, shouldRotate, "stderr.20 should rotate")
 
 	shouldRotate = CheckFileForRotation("propsapi_i-0827338c_tomcat_catalina.out_20160414_0124.log", nil)
-	if !shouldRotate {
-		t.Fatal("propsapi_i-0827338c_tomcat_catalina.out_20160414_0124.log should rotate")
-	}
+	assert.True(t, shouldRotate, "propsapi_i-0827338c_tomcat_catalina.out_20160414_0124.log should rotate")
 
 	shouldRotate = CheckFileForRotation("tomcat-startup.log", nil)
-	if shouldRotate {
-		t.Fatal("tomcat-startup should not rotate")
-	}
+	assert.False(t, shouldRotate, "tomcat-startup should not rotate")
 
 	nqCore := "nodequark.core.13213.3213"
 	shouldRotate = CheckFileForRotation(nqCore, nil)
-	if !shouldRotate {
-		t.Fatalf("%s should rotate", nqCore)
-	}
+	assert.True(t, shouldRotate, "%s should rotate", nqCore)
 
 	nqBadCore := "nodequark.core.13213.abb.3213"
 	shouldRotate = CheckFileForRotation(nqBadCore, nil)
-	if shouldRotate {
-		t.Fatalf("%s should not rotate", nqBadCore)
-	}
+	assert.False(t, shouldRotate, "%s should not rotate", nqBadCore)
+
+	shouldRotate = CheckFileForRotation("catalina_20200330_21.out", nil)
```

Comment:
Does it make sense to follow the pattern above:

```go
catalainaOut := "catalina_20200330_21.out"
shouldRotate = CheckFileForRotation(catalinaOut, nil)
assert.True(t, shouldRotate, "%s should rotate")
```

Reply:
I started down that road, realized that I should turn this into a table-based test, and then decided that the diff was getting too big.  I'll do the larger change as part of another commit.